doc/nrf/ug_thread_commissioning.rst

```diff
@@ -0,0 +1,211 @@
+
+After the petition is accepted by the Leader:
+
+* In the external commissioning scenario, the connection is established and all subsequent communication between the Commissioner and other Thread devices is done through the Border Agent.
+* In both scenarios, the new Commissioner becomes the only authorized Commissioner.
+* In both scenarios, a periodic message is sent to keep the secure commissioning session open.
+
+.. _thread_ot_commissioning_phases_joining:
+
+Joining
+=======
+
+Joining concerns the Joiner role.
+
+Joining occurs in both commissioning scenarios.
+It involves the following phases:
+
+* The Joiner, that is a potential new device in the Thread network, sends a Discovery Request message on every channel.
+* The Joiner Router receives the message and answers with the Discovery Response message.
+  This message contains network identifiers and Steering Data in the payload.
+* The Joiner uses the information received from the Joiner Router to discover the correct network to connect to.
+
+After the Joiner received the payload from the Joiner Router:
+
+* The connection is established.
+* The secure communication session continues, with a periodic message sent to keep it open.
+
+.. _thread_ot_commissioning_roles_authentication:
+
+Security, authentication, and credentials
+*****************************************
+
+To avoid a situation in which rogue devices join the Thread network, the communication between Commissioner and Joiner (in both scenarios), and Commissioner and Border Agent (in external commissioning) is secured with the Datagram Transport Layer Security (DTLS) authentication protocol session.
+The session is established automatically.
+
+Also the communication between Joiner and Joiner Router is secured, but only when Joiner Router sends network credentials to Joiner using one time key generated by the Commissioner.
+
+During commissioning, the on-mesh Thread Commissioner possesses the network master key by default, while the external Thread Commissioner never gains possession of the network master key.
+
+The commissioning uses the following passwords and credentials:
+
+* Commissioning Credential - Passphrase known by the Leader and shared with the Commissioner of the network.
+* Commissioning Key (PSKc) - Pre-Shared Key for the Commissioner based on the Commissioning Credential which is used to establish the Commissioner Session between the Commissioner and Border Agent.
+  All devices in the Thread network store the PSKc.
+* Joining Device Credential (PSKd) - Passphrase for authenticating a new Joiner device, used to establish a secure session between the Commissioner and the Joiner.
+  When encoded in binary, this passphrase is referred to as Pre-Shared Key for the Device.
+
+  .. note::
+        The credential is composed of at least 6 and no more than 32 uppercase alphanumeric characters (0 to 9 and A to Y, with the exclusion of I, O, Q, and Z).
```

Comment:
Which passphrase?
The Joining Device Credential.

When it comes to the Commissioning Credential there are different requirements:
utf-8 format, 6 bytes minimum, 255 bytes max, no character exclusions.

BTW. why was the  `base32-thread` removed the from the text? IMO it was a good reference.

Reply:
This appears under Joining D. Cred. I'll add a separate note for Commissioning Cred.

Reply:
Great!